clang/runtime/dpct-rt/include/dpct/dpl_extras/dpcpp_extensions.h

```diff
@@ -566,6 +566,89 @@ class radix_sort {
   uint8_t *_local_memory;
 };
 
+/// Load linear segment items into block format across threads
+/// Helper for Block Load
+enum load_algorithm {
+
+  BLOCK_LOAD_DIRECT,
+  BLOCK_LOAD_STRIPED,
+  // To-do: BLOCK_LOAD_WARP_TRANSPOSE
+
+};
+
+// loads a linear segment of workgroup items into a subgroup striped
+// arrangement. Created as free function until exchange mechanism is
+// implemented.
+// To-do: inline this function with BLOCK_LOAD_WARP_TRANSPOSE mechanism
+template <size_t GROUP_WORK_ITEMS, size_t ITEMS_PER_WORK_ITEM, typename InputT,
+          typename InputIteratorT, typename Item>
+__dpct_inline__ void
+load_subgroup_striped(const Item &item, size_t linear_tid,
+                      InputIteratorT block_itr,
+                      InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+  // This implementation does not take in account range loading across
+  // workgroup items To-do: Decide whether range loading is required for group
+  // loading
+
+  size_t subgroup_offset = item.get_sub_group().get_local_range()[0];
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    new (&items[idx]) InputT(
+        block_itr[subgroup_offset + linear_tid + (idx * GROUP_WORK_ITEMS)]);
```

Comment:
With my understanding that this is a work-group level algorithm, I believe the stride here should be the subgroup size instead of `GROUP_WORK_ITEMS`. 

I think we should also compute the initial offset for a subgroup to start as `(sg_id * ITERS_PER_WORK_ITEM * sg_sz + subgroup_offset)` where `sg_sz` is the number of elements in the subgroup and `sg_id` is the subgroup's index in the workgroup (can be queried with `item.get_sub_group().get_global_range()`). If we call this `start`, then I believe the offset within your loop would be computed as `start + idx * sg_sz`.

Reply:
Yes updated.